Add roki-glfw & sample

kdaic opened the pull request:
Regarding issue #6, Added a function to use GLFW as an alternative to GLUT. I tried to make the functionality the same as GLUT.
The sample is example/glfw/glfw_solid_test.c. This is just an imitation of example/glut/glut_solid_test.c. However, this sample lacks more examples than the original GLUT one.  
Please confirm.

zhidao commented:
I thought it would be better to have common parameters (including a pointer to the default camera) and share them for GLUT and GLFW. Please check src/rkgl_camera.c in the latest version of the main branch. I think you can modify your roki_glfw.c by referring the new roki_glut.c.
Regarding glfw_solid_test.c, rkglBox() is available. See example/shape/box_test.c. Also, why don't you test your rkgl*FuncGLFW() in the program?

kdaic commented:
roki_glfw was also updated according to the update of rkgl_camera.  
I also used the rkgl_input global variable and updated the callback function.

kdaic commented on include/roki_gl/roki_glfw.h:
```diff
@@ -0,0 +1,30 @@
+/* RoKi-GL - Robot Kinetics library: visualization using OpenGL
```
hoge

kdaic commented on src/roki_glfw.c:
```diff
@@ -77,13 +78,13 @@ void rkglKeyFuncGLFW(GLFWwindow *window, int key, int scancode, int action, int
   /* get modifier */
   ctrl_key = ( glfwGetKey( window, GLFW_KEY_LEFT_CONTROL ) & GLFW_PRESS ) ? GLFW_KEY_LEFT_CONTROL : 0;
-  rkglMouseStoreInput( button, state, GLFW_PRESS, x, y, ctrl_key );
+  rkglMouseStoreInput( button, state, GLFW_PRESS, rkgl_mouse_x, rkgl_mouse_y, ctrl_key );
```
@zhidao  
I Fixed rkglMouseFuncGLFW().   
I define new function ``rkglMouseSotreButtonMod()`` which stores only button( when GLFW_PRESS is enable ) and modifier.   
This function is suggestion for rkgl_input.h/c.

zhidao commented:
I merged this into the main branch with some modifications.
A major modification is to rename rkglWindowCreateGLFW() to rkglWindowOpenGLFW(), and add another function rkglWindowCreateAndOpenGLFW() in order to avoid confusions.
If you agree with this, I'll close this PR.

kdaic commented:
@zhidao  
Thanks your merging.  
I agree with the modification that rename rkglWindowOpenGLFW() and add rkglWindowCreateAndOpenGLFW().  
It's OK to close this PR.
